File: Doingwhiledothat.js
const row = 5;
let i = 0;

do {
    let str = "";
    let k = 0;
    
    do {
        str += "*";
        k++;
    } while (k < i + 1);
    
    console.log(str);
    i++;
} while (i < row);

File: Doingwhiledothat.py
row = 5
i = 0

while True:  
    line = ""
    k = 0
    
    while True: 
        line += "*"
        k += 1
        if not (k < i + 1):
            break
            
    print(line)
    
    i += 1
    if not (i < row):
        break